tests/integrationv2/processes.py

```diff
@@ -13,7 +201,7 @@ class ManagedProcess(threading.Thread):
     The stdin/stdout/stderr and exist code a monitored and results
     are made available to the caller.
     """
-    def __init__(self, cmd_line, provider_set_ready_condition, timeout, data_source=None):
+    def __init__(self, cmd_line, provider_set_ready_condition, ready_to_send=None, timeout=5, data_source=None):
```

Comment:
What unit of time does timeout hold? And if its in minutes, is 5 minutes enough?

Reply:
In Python you almost always default to seconds. The timeout here is seconds (which matches the timeout parameters to all other subprocess timeout calls).

5 seconds is just a default, but seems to be plenty. Most tests succeed in a second. The dynamic record test takes almost 2 seconds because tcpdump takes a little additional time to start up.

If a test takes longer (let's say a test sending a gig of data), then we just increase the timeout.